test/performance/benchmarks/load-test/main.go

```diff
@@ -183,5 +184,19 @@ func checkSLA(results *vegeta.Metrics) error {
 		return fmt.Errorf("SLA 3 failed. Errors occurred: %d", len(results.Errors))
 	}
 
+	// SLA 4: making sure the defined vegeta rates/total requests are met
+	if len(pacers) != len(durations) {
+		return fmt.Errorf("SLA 4 failed. len(pacers)= %s, len(durations)= %s ", strconv.FormatInt(int64(len(pacers)), 10), strconv.FormatInt(int64(len(durations)), 10))
+	}
+	var expectedRequests uint64
+	for i := 0; i < len(pacers); i++ {
+		expectedRequests = expectedRequests + uint64(pacers[i].Rate(durations[i]))
```

Comment:
From what I read this counts the sum of rates per second not the total requests. This needs to be multiplied by the attack duration?